nixos/modules/services/networking/quassel.nix

```
@@ -10,89 +10,406 @@ let
 in
 
 {
-
-  ###### interface
-
   options = {
-
     services.quassel = {
-
-      enable = mkEnableOption (lib.mdDoc "the Quassel IRC client daemon");
-
-      certificateFile = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        description = lib.mdDoc ''
-          Path to the certificate used for SSL connections with clients.
-        '';
-      };
-
-      requireSSL = mkOption {
-        type = types.bool;
-        default = false;
-        description = lib.mdDoc ''
-          Require SSL for connections from clients.
-        '';
-      };
+      enable = mkEnableOption ("the Quassel IRC client daemon");
 
       package = mkOption {
         type = types.package;
         default = pkgs.quasselDaemon;
         defaultText = literalExpression "pkgs.quasselDaemon";
-        description = lib.mdDoc ''
+        description = ''
           The package of the quassel daemon.
         '';
       };
 
-      interfaces = mkOption {
-        type = types.listOf types.str;
-        default = [ "127.0.0.1" ];
-        description = lib.mdDoc ''
-          The interfaces the Quassel daemon will be listening to.  If `[ 127.0.0.1 ]`,
-          only clients on the local host can connect to it; if `[ 0.0.0.0 ]`, clients
-          can access it from any network interface.
+      user = mkOption {
+        default = null;
+        type = types.nullOr types.str;
+        description = ''
+          The existing user the Quassel daemon should run as. If left empty, a default "quassel" user will be created.
         '';
       };
 
-      portNumber = mkOption {
-        type = types.port;
-        default = 4242;
-        description = lib.mdDoc ''
-          The port number the Quassel daemon will be listening to.
-        '';
-      };
+      environmentFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Path to an environment file loaded for the quassel service.
 
-      dataDir = mkOption {
-        default = "/home/${user}/.config/quassel-irc.org";
-        defaultText = literalExpression ''
-          "/home/''${config.${opt.user}}/.config/quassel-irc.org"
-        '';
-        type = types.str;
-        description = lib.mdDoc ''
-          The directory holding configuration files, the SQlite database and the SSL Cert.
+          This can be used to securely store tokens and secrets outside of the world-readable Nix store.
+          Since this file is read by systemd, it may have permission 0400 and be owned by root.
```

I think this should be a note

```suggestion

          ::: {.note}
          Since this file is read by systemd, it may have permission `0400` and be owned by root.
          :::
```